docs/adding-flags.md

```diff
@@ -0,0 +1,105 @@
+# Adding Flags
+
+Adding a flag to **yctf** is easy. While the project is written using the [gin-gonic/gin](https://github.com/gin-gonic/gin) framework, you do not need extensive knowledge using the framework to add a flag. That said, some knowledge is beneficial for you to be able to create unique flags in your handlers. 
+
+Each flag will require five additions to the cod base.
+
+## Adding Metadatda
+
+Each flag has an entry in the `flags.json` file in the following format.
+
+```json
+[
+    ...
+    {
+       "ID": 0,
+       "Flag": "yctf-5f39410913126cea50a3e6fcdb7312628a869da376985fdb7ef7d0a41254a0e3", 
+       "Template": "flag0.html"
+    }
+    ...
+]
+```
+
+- `ID` is the flag identifier. This must be unique.
+- `Flag` is a random string with the `yctf-` prefix that is to be found by the participant. This must be unique.
```

Comment:
```suggestion
- `Flag` is a sha-2 (with a digest of 256) string with the `yctf-` prefix that is to be found by the participant. This must be unique. The sha-2 string is the hash of `yctf-ID`.
```

Reply:
Can you clarify the last piece?
> The sha-2 string is the has of `yctf-ID`.

Would it make sense to include a small pkg that can be run to spit out a hash on an input? Maybe using `go run pkg/generateflag/ ${id}`?

Either way, I've changed this wording.

Reply:
Adding a function/pkg for that is part of #4 feel free to tackle that one too :smile_cat: